src/st_stuff.c

```diff
@@ -718,6 +744,39 @@ void ST_updateWidgets(void)
         keyboxes[i] = keyboxes[i]==-1 || sts_traditional_keys ? i+3 : i+6;
     }
 
+  // [crispy] blinking key or skull in the status bar
+  if (plyr->keyblinktics)
+  {
+    if (!STRICTMODE(hud_blink_keys) || st_classicstatusbar || !hud_displayed)
+    {
+      plyr->keyblinktics = 0;
+    }
+    else
+    {
+      if (!(plyr->keyblinktics & (2*KEYBLINKMASK - 1)))
+        S_StartSound(NULL, sfx_itemup);
+
+      plyr->keyblinktics--;
+
+      for (i = 0; i < 3; i++)
+      {
+        if (!plyr->keyblinkkeys[i])
+          continue;
+
+        keyboxes[i] = plyr->keyblinktics & KEYBLINKMASK
+                      ? plyr->keyblinkkeys[i] == KEYBLINK_BOTH
+                        ? i + 6
+                        : (plyr->keyblinkkeys[i] == KEYBLINK_SKULL
+                           || (plyr->keyblinkkeys[i] == KEYBLINK_EITHER
+                               &&  (plyr->keyblinktics & (2*KEYBLINKMASK))
+                               && !(plyr->keyblinktics & (4*KEYBLINKMASK))))
+                          ? i + 3
+                          : i
+                      : -1;
```

Comment:
Wow, this is almost Lee Killough style. Maybe reformat it? For example:
```c
switch (plyr->keyblinkkeys[i])
{
    case KEYBLINK_BOTH:
        ...
        break;
    case KEYBLINK_SKULL:
        ...
        break;
}
```

Reply:
Is a60c83a3bec77cc5e1dcfa0cc99cc4bd24cad7f9 good?

Reply:
Yes, thanks!